test/Libraries/Microsoft.Extensions.Http.Telemetry.Tests/Logging/Internal/TestLoggingHandler.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Net.Http;
+using Microsoft.Extensions.Http.Logging;
+using Xunit;
+
+namespace Microsoft.Extensions.Http.Telemetry.Test.Logging.Internal;
+
+internal sealed class TestLoggingHandler : DelegatingHandler
+{
+    [SuppressMessage("Reliability", "CA2000:Dispose objects before losing scope", Justification = "Base class disposes the handler")]
+    public TestLoggingHandler(IHttpClientLogger logger, HttpMessageHandler? innerHandler)
+        : base(CreateInternalBclLoggingHandler(logger, innerHandler))
+    {
+    }
+
+    private static DelegatingHandler CreateInternalBclLoggingHandler(IHttpClientLogger logger, HttpMessageHandler? innerHandler)
+    {
+        var handlerType = typeof(IHttpClientFactory).Assembly.GetType("Microsoft.Extensions.Http.Logging.HttpClientLoggerHandler");
```

Comment:
This was taken from Natalia's PoC branch https://github.com/dotnet/extensions/compare/main...CarnaViire:extensions:poc-hcf-custom-logging

Reply:
Are you sure you need this? The intent on my side was just to show that our HttpClientLoggerHandler passes all the tests you had for your logging handler. Like, "you can trust our handler" 😄

So I believe the set of tests that were verifying the handler itself can be removed from dotnet/extensions repo (I haven't looked at the PR yet, I don't know how exactly you've transformed the case).

Or if you do need a handler for some other tests, create it "normal way" via IHttpMessageHandlerFactory.